### ISSUE TITLE
How to use Metrics class to check the performance of the trained model

### ISSUE THREAD

**Issue by tugot17** [airctic/icevision#369]
## 📓 New <Tutorial/Example>
**Is this a request for a tutorial or for an example?**  
Tutorial

**What is the task?**
Lets say I trained a model and now I want to check its performance on the brand new dataset. I don't want to retrain the whole model, I just want to check its performance. How to do it using Metric Class. Lets  assume that COCO metric is fine for me. 

```python
metric = COCOMetric(metric_type=COCOMetricType.bbox)
```

How do I use it so I know exact performance on the new dataset?

---
**Don't remove**
Main issue for examples: #39

**Comment by lgvaz** [comment]
Actually, there is no current easy way of doing this, can you describe the solution you would like?

Maybe a function that take the model, a dataloader, and the metrics?

**Comment by tugot17** [comment]
Maybe a list of metrics instead of a single metric?

**Comment by FraPochetti** [comment]
@fstroth @potipot you guys both worked on COCOMetric.
Is this possible?
Frederik, maybe with your dashboard?

**Comment by fstroth** [comment]
@FraPochetti The dashboard lib already allows for evaluating a model on new (validation) data outside of training.

**Comment by FraPochetti** [comment]
Great. Closing this one, then.
@tugot17 please refer to the [IceVision Dashboard](https://github.com/fstroth/icevision_dashboards) built by @fstroth  for more info.

**Comment by potipot** [comment]
There is an option to test a trained model. Assuming its weights are already loaded you can use
for fastai: 
```python
learner.validate() 
```
for pytorch-lightning
```python
result = trainer.test(test_dl)
```

**Comment by FraPochetti** [comment]
> There is an option to test a trained model. Assuming its weights are already loaded you can use for fastai:
> 
> ```python
> learner.validate(test_dl) 
> ```
> 
> for pytorch-lightning
> 
> ```python
> result = trainer.test(lightning_model, test_dl)
> ```

Would this calculate the COCOMetric?
I guess it would compute any metric the learner (or PL Module) was instantiated with.

**Comment by potipot** [comment]
I'd say this could be a FAQ so we should think of showcasing this in a notebook linked on a website.

**Comment by FraPochetti** [comment]
Yes I agree. 
Keeping it here open as a reminder

**Comment by FraPochetti** [comment]
@ai-fast-track maybe we can add a cell or two showing how to do this to the `inference.ipynb`

**Comment by AlexandreBrown** [comment]
@potipot @FraPochetti 
```python
result = trainer.test(lightning_model, test_dl)
```  
This does not work in PL right now since IceVision PL adapters do not implement the `test_step`
```
pytorch_lightning.utilities.exceptions.MisconfigurationException: No `test_step()` method defined to run `Trainer.test`
```

**Comment by AlexandreBrown** [comment]
Just opened a PR to add support for Pytorch Lightning Trainer test with IceVision models.

Take a look https://github.com/airctic/icevision/pull/1125

This should solve your issue.
